src/App.tsx

```diff
@@ -1,7 +1,38 @@
-import React from 'react';
+import { PureComponent } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string;
+};
+
+export class App extends PureComponent<{}, State> {
+  state: Readonly<State> = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener(
+      'keyup',
+      this.handleKeyup,
+    );
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  handleKeyup = (event: KeyboardEvent) => this.setState({ key: event.key });
+
+  render() {
+    return (
+      <div className="App">
+        <p className="App__message">
+          {this.state.key ? (
+            `The last pressed key is [${this.state.key}]`
```

Comment:
You can perform destructuring of this key. It's not necessary in this case, but if you'll had more that one variable it should be good (i think you know that) )